bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/api/AgsMagnitude.scala

```diff
@@ -31,10 +28,8 @@ object AgsMagnitude {
    * Gets the widest possible range limits incorporating the given conditions
    * and speeds.
    */
-  def rangeLimits(mc: MagnitudeCalc, c1: (Conditions, GuideSpeed), c2: (Conditions, GuideSpeed)): MagnitudeConstraints = {
-    val ml = mc(c1._1, c1._2).union(mc(c2._1, c2._2)).getValue
-    MagnitudeConstraints(ml.getBand.toNewModel, FaintnessConstraint(ml.getFaintnessLimit.getBrightness), ml.getSaturationLimit.asScalaOpt.map(s => SaturationConstraint(s.getBrightness)))
-  }
+  def rangeLimits(mc: MagnitudeCalc, c1: (Conditions, GuideSpeed), c2: (Conditions, GuideSpeed)): Option[MagnitudeConstraints] =
+    mc(c1._1, c1._2).union(mc(c2._1, c2._2))
```

Comment:
The `Option` bubbling up is unfortunate since it shouldn't be possible to end up with a `None` there if we could fix the AGS API.  That isn't the point of this task though so I'm willing to let it slide.